src/shims/fs.rs

```diff
@@ -257,6 +264,197 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         this.try_unwrap_io_result(result)
     }
 
+    fn statx(
+        &mut self,
+        dirfd_op: OpTy<'tcx, Tag>, // Has type `int`
+        pathname_op: OpTy<'tcx, Tag>, // Has type `const char *`
+        flags_op: OpTy<'tcx, Tag>, // Has type `int`
+        _mask_op: OpTy<'tcx, Tag>, // Has type `unsigned int`
+        statxbuf_op: OpTy<'tcx, Tag> // Has type `struct statx *`
```

Comment:
I thought you said it's not actually `struct statx *`, but `isize` or so?

Reply:
I should say "Should have type" instead of "Has type". Everything there is an `isize`.

Reply:
Everything is an `isize`? Well that's weird... I guess syscalls do that...
But yes please clarify that in the comments.

Reply:
I spent 2 hours trying to figure out why the layouts were wrong, then I discovered everything was an integer D:

Reply:
Ouch... yeah the syscall API is entirely untyped on the Rust side so this easily happens.